### PR TITLE
refactor(debug): use simple events search instead of searchLatest

### PR DESCRIPTION
**Pull request by gcusnieux**
gravitee-io/issues#6126

### CODE REVIEW

File: gravitee-gateway-services/gravitee-gateway-services-debug/src/main/java/io/gravitee/gateway/debug/reactor/DebugReactor.java [hunk 1]
```diff
@@ -111,11 +112,11 @@ public void onEvent(Event<ReactorEvent, Reactable> reactorEvent) {
                                     reactableDebugApi.setResponse(response);
                                     debugEvent.setPayload(objectMapper.writeValueAsString(convert(reactableDebugApi)));
                                     updateEvent(debugEvent, ApiDebugStatus.SUCCESS);
+                                    logger.info("Debugging successful, removing the handler.");
```
Comment (ytvnr) [inline gravitee-gateway-services/gravitee-gateway-services-debug/src/main/java/io/gravitee/gateway/debug/reactor/DebugReactor.java:115]: 👍

File: gravitee-gateway-services/gravitee-gateway-services-sync/src/test/java/io/gravitee/gateway/services/sync/synchronizer/OrganizationSynchronizerTest.java [hunk 2]
```diff
@@ -58,7 +58,7 @@
     private GatewayConfiguration gatewayConfiguration;
 
     @Mock
-    private ObjectMapper objectMapper;
+        private ObjectMapper objectMapper;
```
Comment (ytvnr) [inline gravitee-gateway-services/gravitee-gateway-services-sync/src/test/java/io/gravitee/gateway/services/sync/synchronizer/OrganizationSynchronizerTest.java:61]: fail ? or bad indentation in github ?

**Review by ytvnr — APPROVED** [review]
just a remark about indentation

**Review by ytvnr — APPROVED** [review]

**Review by jhaeyaert — CHANGES_REQUESTED** [review]
Here are some highlight points to check:
- Make sure the created httpClient is well disposed in any circonstances, success or error (see: https://github.com/gravitee-io/gravitee-gateway/pull/761/files#diff-693ed57c429dcbbd0b7df8064875f2b7412ea8b6e345eb5a6eede3b23f982ab3R89)
- The connection timeout seems missing on the httpclient (see: https://github.com/gravitee-io/gravitee-gateway/blob/ebe1b1fd533f54c96913d6d7d8180f94797980dc/gravitee-gateway-services/gravitee-gateway-services-debug/src/main/java/io/gravitee/gateway/debug/reactor/DebugReactor.java#L179-L191).
- The different timeouts (connectionTimeout, requestTimeout) should have default values and be configurable. Currently, request timeout is set to 5000ms without possibility to customize it (see https://github.com/gravitee-io/gravitee-gateway/blob/ebe1b1fd533f54c96913d6d7d8180f94797980dc/gravitee-gateway-services/gravitee-gateway-services-debug/src/main/java/io/gravitee/gateway/debug/reactor/DebugReactor.java#L206).
- **[Not really relevant because requests are async so it will not have any impact on other debug requests]** ---> Now that you move to stream (previously a flowable), the events are processed sequentially. It means that if there are multiple debug request to process and some of them take too long, others will be impacted (see: https://github.com/gravitee-io/gravitee-gateway/blob/ebe1b1fd533f54c96913d6d7d8180f94797980dc/gravitee-gateway-services/gravitee-gateway-services-sync/src/main/java/io/gravitee/gateway/services/sync/synchronizer/DebugApiSynchronizer.java#L66)

**Review by jhaeyaert — APPROVED** [review]

**Pull request merged**